ucc listener: some fixes for being able to run all tests (even ones we don't support at the moment)

tomereli opened the pull request:
This PR fixes two issues which currently prevent us from running all agent tests in a single run:

1. Reply `INVALID` if non-eth backhaul onboarding command is received
2. Always reply with `RUNNING` first as it appears that the UCC hangs since it expects RUNNING regardless of the command success/fail criteria

Signed-off-by: Tomer Eliyahu <tomer.b.eliyahu@intel.com>

Commit:
agent: ucc listener: fail wifi bh tests immediately

Since we do not support wifi backhaul at the moment, tests which require
wifi backhaul get stuck and fail eventually since the agent performs
slave resets until it finishes the stop_on_failure counter.
In order to prevent that, add a check for this in the agent ucc handler
and fail the test immediately if the backhaul is not eth.

Signed-off-by: Tomer Eliyahu <tomer.b.eliyahu@intel.com>

vitalybu commented on agent/src/beerocks/slave/agent_ucc_listener.cpp:
```diff
@@ -192,6 +192,13 @@ bool agent_ucc_listener::handle_dev_set_config(std::unordered_map<std::string, s
+    // TODO - add wireless backhaul support.
+    // For now, this causes slave reset and the test to hang,
+    // so return not supported.
+    if (backhaul_param != "eth") {
```
LGTM, but wondering if `backhaul_param` can also be empty..

tomereli replied:
It cannot, its invalid if it is

Commit:
bcl: ucc listener: always reply with status,RUNNING

Apparently the UCC expects to always receive status,RUNNING before any
other status, so even if there is a failure with which we reply with
status,INVALID,* - the UCC ignores this and hangs.

This was observed when the UCC sent a command which is currently not
supported in the ucc listener - start_wps_registration, which resulted
with an immediate INVALID reply which was ignored.

Therefore, always send the RUNNING reply first.

Signed-off-by: Tomer Eliyahu <tomer.b.eliyahu@intel.com>

rmelotte reviewed (approved):
There are some typos in the commit messages: 
immidiate -> immediate
immidietely -> immediately

Otherwise LGTM, so I'm approving to not hold the merge, but removing the merge tag so you can decide when you want it merged.

arnout reviewed (approved):
> Apparently the UCC expects to always receive status,RUNNING before any
> other status, so even if there is a failure with which we reply with
> status,INVALID,* - the UCC ignores this and hangs.

That is so insane... Figure 1 of the Control API Specification document clearly shows that INVALID comes on its own, and ERROR comes after RUNNING.

Given this situation, I think it would be better to never return INVALID at all, and return ERROR instead - that way we at least stay a little closer to the specification, and we work around this UCC issue.

But that really is nitpicking, and I don't think we should spent too much time on this since it's only for certification, so approving.

BTW there are still two INVALID branches that precede the RUNNING: forbidden_chars and cmd_tokens_vec.empty().

tomereli commented:
> BTW there are still two INVALID branches that precede the RUNNING: forbidden_chars and cmd_tokens_vec.empty().

Fixed.

arnout commented:
Note: CI hit #832 (cfr. https://gitlab.com/prpl-foundation/prplMesh/-/jobs/483262491/artifacts/browse/logs/gateway-30394/ - there are two log files for beerocks_agent_wlan0). I manually retriggered the failing test and now it should pass.
